src/megatron/energon/flavors/webdataset/metadata.py

```diff
@@ -41,11 +41,11 @@ def from_config(
         """
         parser = JsonParser(strict=True)
         info = parser.raw_to_typed(
-            yaml.safe_load((path / MAIN_FOLDER_NAME / info_config).read_text()),
+            load_yaml((path / MAIN_FOLDER_NAME / info_config).read_bytes()),
```

Comment:
As agreed, try to load `.info.json` first from now on and fall back to `.info.yaml` if it cannot be loaded. I think we can remove the `info_config` argument.